SS2-Project/Assets/Starstorm2/ScriptableObjects/SS2AssetCollections.cs

```diff
@@ -8,19 +8,70 @@
 using System.Collections.Generic;
 using UnityEngine;
 using UnityEditor;
+using System.Reflection;
+using HG;
+using UnityEngine.Networking;
+using RoR2.Projectile;
+using System.IO;
+using Path = System.IO.Path;
+using System.Linq;
 namespace SS2
 {
-    //todo: a lot
+    public class ExtendedAssetCollection : AssetCollection
+    {
+        public void OnValidate()
+        {
+            //very slow lol
+
+            //ensure field values are contained in AssetCollection.assets
+            foreach (FieldInfo f in this.GetType().GetFields())
+            {
+                object val = f.GetValue(this);
+                if (val == null || (val != null && !(val is UnityEngine.Object))) continue;
+                UnityEngine.Object obj = (UnityEngine.Object)val; // "specified cast is not valid" yes it is dumbfuck
+                if (!obj) continue;
+                bool contains = false;
+                for (int i = 0; i < assets.Length; i++)
+                {
+                    if (assets[i] == obj)
+                    {
+                        contains = true;
+                        break;
+                    }
+                }
+                if (!contains)
+                {
+                    ArrayUtils.ArrayInsert<UnityEngine.Object>(ref this.assets, 0, obj);
+                }
+            }
+
+            //ensure all assets are distinct
+            List<UnityEngine.Object> objects = new List<UnityEngine.Object>();           
```

Comment:
I am pretty sure we can utilize the linq method "Distinct()" which should basically make sure no objects are duplicated.